# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsTextFrame.h
# Commit: c4a471b86fc0
# Full Hash: c4a471b86fc0091946ad142515332c5532be5f62
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2023-03-02 04:57:23
# Regressor Bug: 1818654
# File Overlap Count: 3
# Description:
#   Bug 1818654 - Don't give up on clipping to the visible rect in nsDisplayText when selection or shadow is present. r=emilio
#   
#   Although I haven't been able to reproduce the reporter's OOM crash here, I hope this will avoid
#   the issue; it certainly improves performance characteristics in my local build.
#   
# ==============================================================================

diff -r 80627767254a -r c4a471b86fc0 layout/generic/nsTextFrame.h
--- a/layout/generic/nsTextFrame.h	Wed Mar 01 11:28:54 2023 +0000
+++ b/layout/generic/nsTextFrame.h	Wed Mar 01 11:37:27 2023 +0000
@@ -925,6 +925,16 @@
   bool CombineSelectionUnderlineRect(nsPresContext* aPresContext,
                                      nsRect& aRect);
 
+  // This sets *aShadows to the appropriate shadows, if any, for the given
+  // type of selection.
+  // If text-shadow was not specified, *aShadows is left untouched.
+  // If aTextPaintStyle is provided, it will be used to avoid the need to
+  // instantiate a temporary one.
+  void GetSelectionTextShadow(
+      SelectionType aSelectionType,
+      mozilla::Span<const mozilla::StyleSimpleShadow>* aShadows,
+      nsTextPaintStyle* aTextPaintStyle = nullptr);
+
   /**
    * Utility methods to paint selection.
    */